src/npf/rproc/npf_ext_appfw.c

```diff
@@ -389,10 +389,15 @@ appfw_action(npf_cache_t *npc, struct rte_mbuf **nbuf, void *arg,
 	 */
 	dpi_flow = npf_session_get_dpi(se);
 	if (!dpi_flow) {
+#ifdef USE_NDPI
 		uint8_t engines[] = { IANA_USER, IANA_NDPI };
-
+		size_t engines_len = 2;
+#else
+		uint8_t engines[] = { IANA_USER };
+		size_t engines_len = 1;
```

Comment:
@dfawcus

The second line would be better outwith of the conditional and written as:

    size_t engines_len = ARRAY_SIZE(engines);

Which raises another question, each other call to dpi_session_first_packet() passes an final pair or parameters, being 'engines_length' and 'engines', however 'engines_length' was a constant '2' in each case.

This change implies that something related to that pair should be altered - somehow.